src/fsharp/vs/service.fs

```diff
@@ -2215,11 +2233,36 @@ type BackgroundCompiler(referenceResolver, projectCacheSize, keepAssemblyContent
 
     // Also keyed on source. This can only be out of date if the antecedent is out of date
     let parseAndCheckFileInProjectCache = 
-        MruCache<string * string * FSharpProjectOptions, FSharpParseFileResults * FSharpCheckFileResults * int * DateTime>
+        MruCache<FileName * Source * FSharpProjectOptions, FSharpParseFileResults * FSharpCheckFileResults * FileVersion * DateTime>
             (keepStrongly=incrementalTypeCheckCacheSize,
              areSame=AreSameForChecking3,
              areSameForSubsumption=AreSubsumable3)
 
+    /// Holds keys for files being currently checked. It's used to prevent checking same file in parallel (interliveing chunck queued to Reactor).
+    let beingCheckedFileCache = 
+        System.Collections.Concurrent.ConcurrentDictionary<FilePath * ProjectPath * FileVersion, unit>(
+            { new IEqualityComparer<FilePath * ProjectPath * FileVersion> with
```

Comment:
I've just remove the explicit comparer because tuples work nicely with dictionary.